src/state/experiments.js

```diff
@@ -14,11 +25,27 @@ export const LOAD = 'ca/experiments/LOAD';
 export const LOAD_SUCCESS = 'ca/experiments/LOAD_SUCCESS';
 export const LOAD_ERROR = 'ca/experiments/LOAD_ERROR';
 export const SET = 'ca/experiments/SET';
+export const CREATE = 'ca/experiments/CREATE';
+export const CREATE_SUCCESS = 'ca/experiments/CREATE_SUCCESS';
+export const CREATE_ERROR = 'ca/experiments/CREATE_ERROR';
+export const UPDATE = 'ca/experiments/UPDATE';
+export const UPDATE_SUCCESS = 'ca/experiments/UPDATE_SUCCESS';
+export const UPDATE_ERROR = 'ca/experiments/UPDATE_ERROR';
+export const UPLOAD = 'ca/experiments/UPLOAD';
+export const UPLOAD_SUCCESS = 'ca/experiments/UPLOAD_SUCCESS';
+export const UPLOAD_ERROR = 'ca/experiments/UPLOAD_ERROR';
+export const OPEN_EDIT_MODAL = 'ca/experiments/OPEN_EDIT_MODAL';
```

Comment:
usually, it's a good idea to keep store agnostic to presentation level.

Reply:
To be honest I didn't like that much the previous code, but I didn't want to spend much more time refactoring it. I've tried to improve it in https://github.com/src-d/code-annotation/pull/195/commits/ce357ffaf5ea919e448cbcd238e52032ed6373cd, moving the modal state to Experiments.js